driver-sync/src/test/functional/com/mongodb/client/SessionsTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.event.CommandEvent;
+import com.mongodb.event.CommandStartedEvent;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import util.JsonPoweredTestHelper;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static com.mongodb.JsonTestServerVersionChecker.skipTest;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+// See https://github.com/mongodb/specifications/tree/master/source/sessions/tests
+@RunWith(Parameterized.class)
+public class SessionsTest extends AbstractTransactionsTest {
```

Comment:
Very weird that SessionsTest extends AbstractTransactionsTest.  Should the latter be renamed?  Should a new superclass be introduced?

Reply:
I did it this way because the spec states that the sessions tests use the same operations as the transactions tests with a few additions. The abstract class name is reasonable since the class is primarily associated with transactions tests.

Reply:
In a recent discussion, we decided to leave the name as is until 4.x.